src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.EasyRequest.cs

```diff
@@ -304,6 +311,17 @@ internal void SetCookieOption(Uri uri)
                 }
             }
 
+            private void SetClientCertificateOptions()
+            {
+                if (_clientCertProvider == null)
+                {
+                    return;
+                }
+                SetCurlOption(CURLoption.CURLOPT_SSLCERT, _clientCertProvider._certificateFile);
```

Comment:
`openssl s_server -accept 1234 -www -cert acertandprivatekey.pem -verify 20` will cause https://localhost:1234/ to request a client authentication certificate.

If you want it to require a client authentication certificate, make Verify capitalized: `openssl s_server -accept 1234 -www -cert acertandprivatekey.pem -Verify 20`

(The 20 is how tall the chain is allowed to be, it's just a required parameter for their command-line utility).